### PR TITLE
ngclient: use MetaFile/TargetFile verification

### LINKED ISSUE #1410: ngclient: use MetaFile/TargetFile verification API when it exists
Related to #1377 and #1361: Once Metadata API offers the verification functionality, we should use it in ngclient and verify file lengths and hashes for both metadata and targets.

This would mean removal of helper functions like _check_file_length() _check_hashes_obj(), and removal of hash checking code in metadatabundle

### PR DESCRIPTION
**Pull request by sechkova**
Fixes #1410

**Description of the changes being introduced by the pull request**:

Use the newly added  MetaFile/TargetFile.verify_length_and_hashes in the ngclient workflow.

**Please verify and check that the pull request fulfills the following
requirements**:

- [ ] The code follows the [Code Style Guidelines](https://github.com/secure-systems-lab/code-style-guidelines#code-style-guidelines)
- [ ] Tests have been added for the bug fix or new feature
- [ ] Docs have been added for the bug fix or new feature

### CODE REVIEW

**Comment by sechkova** [comment]
Only the last two commits are new in this PR: 39ebe92  f8ecac8

**Comment by sechkova** [comment]
Rebased on `develop`

File: tuf/ngclient/updater.py [hunk 1]
```diff
@@ -205,17 +205,21 @@ def download_target(
         else:
             target_base_url = _ensure_trailing_slash(target_base_url)
 
-        full_url = parse.urljoin(target_base_url, targetinfo["filepath"])
+        target_filepath = targetinfo["filepath"]
+        target_fileinfo = targetinfo["fileinfo"]
```
Comment (jku) [inline tuf/ngclient/updater.py]:
```suggestion
        target_fileinfo:TargetFile = targetinfo["fileinfo"]
```
the real fix would be fixing the targetinfo argument type but this helps typing in the mean time

**Review by jku — COMMENTED** [review]
Looks good I think. Handling the exception just to raise again is a bit annoying but ... also probably correct:
* in these cases we do want to raise a RepositoryError -- client has no way to "handle" this error
* but other cases might be different (like if client itself was testing a cached file)

just on one question: what about the targetfile check in `updated_targets()`? it's a bit different case but should be able to use the same methods

**Comment by sechkova** [comment]
> just on one question: what about the targetfile check in `updated_targets()`? it's a bit different case but should be able to use the same methods

:roll_eyes: 
1b28604 <- I used them now
aeccb05 is the same commit with added `TargetFile` type

**Review by jku — APPROVED** [review]
:+1:

**Pull request merged**